src/lib/constants/Messages.ts

```diff
@@ -36,4 +36,26 @@ export const Messages = {
       'vtex workspace create <workspace> --production'
     )} to create a production workspace`,
   PROMOTE_SPINNER_START: 'Preparing the workspace to be promoted',
+  PROMOTE_CONFLICT_ERROR: (workspace: string, buckets: string[]) =>
+    `\n  ${chalk.bold(
+      `${chalk.red('Error')}: The workspace ${ColorifyConstants.ID(workspace)} couldn't be promoted ` +
+        `because it has conflicts with ${ColorifyConstants.ID('master')}`
+    )}${
+      buckets.length === 0
+        ? chalk.bold('.')
+        : `${`${chalk.bold(' in the following data:')}\n\n` + `${buckets.map((bucket) => `    ${bucket}`).join('\n')}`}`
+    }` +
+    `\n\n  ` +
+    `${chalk.bold(
+      `You can try to resolve the conflicts above with any request on workspace ` +
+        `${ColorifyConstants.ID(workspace)} that interacts with the conficting data.`
+    )}` +
+    `\n  ` +
+    `If you do so you allow the service that handles the data to resolve any conflicts before the workspace promotion.` +
```

Comment:
"By doing so, you allow the service that handles the data to resolve any conflicts before the workspace promotion." looks more straight to the point.